lib/flutter_command.dart

```diff
@@ -421,6 +447,42 @@ abstract class Command<TParam, TResult> extends ValueNotifier<TResult> {
     return _futureCompleter.future;
   }
 
+  /// Pipes one [Command]'s result [value] to another command as input and
+  /// returns the pipedCommand for further processing like adding another
+  /// [Command] as piped command.
+  ///
+  /// If the [pipedCommandTakesParam] is set to false then [pipedCommand] is
+  /// called without passing any parameters.
+  ///
+  /// If the [pipedCommandTakesParam] is set to true [pipedCommand] is called
+  /// with the resulting [value] of the previous command only if the
+  /// [pipedCommand] accepts parameters of the same type as [TPipedCommandParam].
+  ///
+  /// As a fall back case, [pipedCommandParam] will be used if the [value]
+  /// returned by the command doesn't match the parameter type of the
+  /// [pipedCommand]. This allows connecting non compatible commands if needed.
+  ///
+  Command pipeResult<TPipedCommandParam>(
+    Command pipedCommand, {
+    @required bool pipedCommandTakesParam,
+    TPipedCommandParam pipedCommandParam,
+    Duration debounceDuration = const Duration(),
+  }) {
+    assert(pipedCommand != null, 'pipedCommand cannot be null');
+    assert(pipedCommandTakesParam != null,
+        'pipedCommandTakesParam cannot be null');
+    debounce(debounceDuration).listen((TResult resultValue, _) {
```

Comment:
where does this `debounce` come from?
Oh wait, you used the debounce extension. please write `this.debounce` I really took a moment.

Reply:
I was getting a dart lint hint for this saying unnecessary this.

```
Don't access members with `this` unless avoiding shadowing. dart(unnecessary_this)
```

Reply:
I was getting a dart lint hint saying unnecessary this. The debounce comes from functional_listener I believe.

```
Don't access members with `this` unless avoiding shadowing. dart(unnecessary_this)
```

Reply:
yeah, please disable this lint in analysis_option. sometimes especially in cases like this it's confusing not to use `this.` if you use an extension method. It took me several minutes to finally get it